fendermint/vm/genesis/src/lib.rs

```diff
@@ -95,6 +95,45 @@ pub struct Actor {
     pub balance: TokenAmount,
 }
 
+/// Total power from IPC LibStaking, consists of the collateral staked by the validator and the
+/// federated power assigned by the super admin/owner of the subnet
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct LibStakingPower {
+    #[serde_as(as = "IsHumanReadable")]
+    pub collateral: TokenAmount,
+    #[serde_as(as = "IsHumanReadable")]
+    pub federated_power: TokenAmount,
+}
+
+impl LibStakingPower {
+    /// Convert from [LibStakingPower] to [Power] by specifying the number of significant
+    /// decimal places per FIL that grant 1 power.
+    ///
+    /// For example:
+    /// * with 3 decimal places, we get 1 power per milli FIL: 0.001 FIL => 1 power
+    /// * with 0 decimal places, we get 1 power per whole FIL: 1 FIL => 1 power
+    pub fn into_power(self: LibStakingPower, scale: PowerScale) -> Power {
+        let atto_per_power = Collateral::atto_per_power(scale);
+        let atto = self.federated_power.atto() + self.collateral.atto();
+        // Rounding away from zero, so with little collateral (e.g. in testing)
+        // we don't end up with everyone having 0 power and then being unable
+        // to produce a checkpoint because the threshold is 0.
+        let power = atto.div_ceil(&atto_per_power);
+        let power = power.min(BigInt::from(u64::MAX));
+        Power(power.try_into().expect("clipped to u64::MAX"))
+    }
```

Comment:
There shouldn't be a need to copy-paste all this. The original is covered by tests, this isn't. 

Since both `federated_power` and `collateral` are `TokenAmount`, you can just add the together, wrap the result in `Collateral` and call `into_power` on it: `Collateral(self.federated_power + self.collateral).into_power(scale)`.